packages/wdio-appium-service/src/launcher.js

```diff
@@ -1,46 +1,46 @@
 import logger from '@wdio/logger'
 import { spawn } from 'child_process'
 import { createWriteStream, ensureFileSync } from 'fs-extra'
-import { paramCase } from 'param-case'
 import { promisify } from 'util'
-import getFilePath from './utils/getFilePath'
+import { getFilePath, getAppiumCommand, cliArgsFromKeyValue } from './utils'
 
 const log = logger('@wdio/appium-service')
 const DEFAULT_LOG_FILENAME = 'appium.txt'
 
 export default class AppiumLauncher {
-    constructor() {
-        this.logPath = null
-        this.command = ''
+    constructor(options, caps, config) {
+        this.options = options
+        this.logPath = options.logPath || config.outputDir
```

Comment:
Is there a chance that both of those would be undefined?

Reply:
yes, in this case we would not print the Appium logs into a file.